### PR TITLE
Avoid IPv4 when unsupported

### PR DESCRIPTION
**Pull request by natebosch**

### CODE REVIEW

**Comment by grouma** [comment]
I'm confused by this change. What's the intent? Should we conditionally choose the hostname based off support?

File: test/http_multi_server_test.dart [hunk 1]
```diff
@@ -296,26 +296,18 @@ void main() {
...
 
     test('listens on specified hostname', () async {
-      if (await supportsIPv4) {
-        final server = await HttpMultiServer.bindSecure(
-            InternetAddress.anyIPv4, 0, context);
-        server.listen((request) {
-          request.response.write('got request');
-          request.response.close();
-        });
+      if (!await supportsIPv4) return;
```
Comment (grouma) [inline test/http_multi_server_test.dart:299]: I believe some environments can support IPv6 and not IPv4. We shouldn't eagerly return.

Comment (natebosch) [reply test/http_multi_server_test.dart:299]: I used the same pattern as the similar test for the insecure bind.

https://github.com/dart-lang/http_multi_server/blob/7aca9e87d4a68374b685334f20359320054b8d7b/test/http_multi_server_test.dart#L234-L235

I'm not entirely sure why we landed on that pattern. https://github.com/dart-lang/http_multi_server/pull/30

I think the reason might be that checking the `anyIPV4` is the fastest shortcut for this check that it _isn't_ listening on an IPv6 address.

Comment (grouma) [reply test/http_multi_server_test.dart:299]: We could optionally bind to `anyIPv6` and update the expects accordingly but it probably doesn't really matter.

**Review by grouma — APPROVED** [review]

File: test/http_multi_server_test.dart (hunk 1 above)
Comment (natebosch) [reply test/http_multi_server_test.dart:299]: IIRC the problem with `anyIPv6` is it also binds an IPv4 address, while the reverse isn't true.

Comment (grouma) [reply test/http_multi_server_test.dart:299]: Does it fail if it can't bind to IPv4?

Comment (natebosch) [reply test/http_multi_server_test.dart:299]: No it doesn't fail, but we don't have a good counterfactual to check against - we can't check that it _didn't_ bind the IPv4 address.

Comment (grouma) [reply test/http_multi_server_test.dart:299]: Got it.

**Pull request merged**